src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -61,20 +106,50 @@ const collectorSignalsLoader = async (
 
   // Handle signals for auction artworks
   if (isInSale && auctionsCollectorSignalsEnabled) {
-    const activeSaleArtwork = await getActiveSaleArtwork(
+    const activeLotData = await getActiveSaleArtwork(
       {
         artworkId,
         saleIds: artwork.sale_ids,
       },
       ctx
     )
 
-    if (activeSaleArtwork) {
+    const activeAuctionLot = activeLotData && !activeLotData.sale.ended_at
+    if (activeAuctionLot) {
+      const { saleArtwork, sale } = activeLotData
+
       if (artwork.recent_saves_count) {
         lotWatcherCount = artwork.recent_saves_count
       }
-      if (activeSaleArtwork.bidder_positions_count) {
-        bidCount = activeSaleArtwork.bidder_positions_count
+
+      if (saleArtwork.bidder_positions_count) {
+        bidCount = saleArtwork.bidder_positions_count
+      }
+
+      const futureRegistrationEndTime = ifFutureDate(sale.registration_ends_at)
+      if (futureRegistrationEndTime) {
+        registrationEndsAt = futureRegistrationEndTime
+      }
+
+      const saleLiveStartAt = sale.live_start_at
+      if (saleLiveStartAt) {
+        const futureLiveStartTime = ifFutureDate(saleLiveStartAt)
+
+        if (futureLiveStartTime) {
+          liveBiddingStarted = false
+          liveStartAt = futureLiveStartTime
+        } else {
+          liveBiddingStarted = true
+        }
+      }
+
+      const extendedBiddingEndAt = saleArtwork.extended_bidding_end_at
+      if (extendedBiddingEndAt && extendedBiddingEndAt.length > 0) {
+        onlineBiddingExtended = true
+        lotClosesAt = extendedBiddingEndAt
+      } else {
+        onlineBiddingExtended = false
+        lotClosesAt = saleArtwork.end_at
```

Comment:
lots of hidden logic and assumptions in here:
- `sale.liveStartAt` is not on sale artwork, and implies the non-existence of `end_at`
- `saleArtwork.extendedBiddingEndAt` implies online bidding was extended and `end_at` should be ignored
- `ifFutureDate()` is a filtering helper that returns a date object if it is (present and) in the future, otherwise null.
- At the top of this section everything is wrapped in an additional check for `!sale.ended_at` to ignore artworks with an ended sale, though [technically we should not get these](https://github.com/artsy/metaphysics/pull/5889/files?diff=split&w=1#diff-dc52d603a602c819aef005d759243d63b275674ea6954b70a204fec0f49d29b5R195) - in the case below we query for sales with the `live: true` param, refers to the auction's active bidding period, not live auction integration.

Reply:
Maybe we can make the assumption and condition clear in the GraphQL field description?